### PR TITLE
Add note for using Git with Self-signed repositories.

### PR DESCRIPTION
**Pull request by smk4664**
### Fixes: #642 

Documentation updates on how to ignore Git SSL verification.

### CODE REVIEW

File: nautobot/docs/models/extras/gitrepository.md [hunk 1]
```diff
@@ -25,6 +25,10 @@ Whenever a Git repository record is created, updated, or deleted, Nautobot autom
 !!! important
     The repository branch must exist and have a commit against it. At this time, Nautobot will not initialize an empty repository.
 
+!!! note
+    If you are using a self-signed git repository, you will need to set the environment variable GIT_SSL_NO_VERIFY="1"
```
Comment (glennmatthews) [inline nautobot/docs/models/extras/gitrepository.md:29]:
```suggestion
    If you are using a self-signed Git repository, you will need to set the environment variable `GIT_SSL_NO_VERIFY="1"`
```

File: nautobot/docs/user-guides/git-data-source.md [hunk 2]
```diff
@@ -79,6 +78,10 @@ Fill out the details for the Git repository. More information on the inputs can
 
 As soon as you click on **Create**, Nautobot will clone and sync the repository and provide status of the job.
 
+!!! note
+    If you are using a self-signed git repository, you will need to set the environment variable GIT_SSL_NO_VERIFY="1"
```
Comment (glennmatthews) [inline nautobot/docs/user-guides/git-data-source.md:82]:
```suggestion
    If you are using a self-signed Git repository, you will need to set the environment variable `GIT_SSL_NO_VERIFY="1"`
```

File: nautobot/docs/user-guides/git-data-source.md [hunk 3]
```diff
@@ -79,6 +78,10 @@ Fill out the details for the Git repository. More information on the inputs can
 
 As soon as you click on **Create**, Nautobot will clone and sync the repository and provide status of the job.
 
+!!! note
```
Comment (glennmatthews) [inline nautobot/docs/user-guides/git-data-source.md:81]: This isn't ideal as a note in a *user* guide, IMHO - the user may not have the ability to set environment variables for Nautobot. I'm not sure how better to phrase it though, perhaps something like "you will need make sure that the Nautobot installation has set the environment variable..."?

File: development/dev.env [hunk 4]
```diff
@@ -28,3 +28,6 @@ REDIS_PASSWORD=decinablesprewad
 # Needed for Selenium integration tests
 NAUTOBOT_SELENIUM_URL=http://selenium:4444/wd/hub  # WebDriver (Selenium client)
 NAUTOBOT_SELENIUM_HOST=nautobot  # LiveServer (Nautobot server)
+
+# Allow self signed git repositories for config contexts, export templates, ...
+# GIT_SSL_NO_VERIFY="1"
```
Comment (glennmatthews) [inline development/dev.env:33]: Might it make sense to document this setting in `docs/configuration/optional-settings.md`?

File: nautobot/docs/user-guides/git-data-source.md (hunk 3 above)
Comment (itdependsnetworks) [reply nautobot/docs/user-guides/git-data-source.md:81]: `If you are using a self-signed git repository, please refer to the <insert link> development guide.` ?

File: development/dev.env (hunk 4 above)
Comment (smk4664) [reply development/dev.env:33]: Ah yes, that would make sense.  I will be updating my PR today.

File: nautobot/docs/configuration/optional-settings.md [hunk 5]
```diff
@@ -367,6 +366,15 @@ The value of this variable can also be customized by setting the environment var
 
 ---
 
+## GIT_SSL_NO_VERIFY
+
+Default: Unset
+
+If you are using a self-signed git repository, you will need to set the environment variable `GIT_SSL_NO_VERIFY="1"`
+in order for the repository to sync.
```
Comment (glennmatthews) [inline nautobot/docs/configuration/optional-settings.md:374]: This *has* to be an environment variable, not a `nautobot_config.py` variable, correct?

```suggestion
in order for the repository to sync.

!!! warning
    This *must* be specified as an environment variable. Setting it in `nautobot_config.py` will not have the desired effect.
```

File: nautobot/docs/user-guides/git-data-source.md [hunk 6]
```diff
@@ -79,6 +78,9 @@ Fill out the details for the Git repository. More information on the inputs can
...
+    If you are using a self-signed git repository, the Server Administrator will need to ensure the [GIT_SSL_NO_VERIFY](../configuration/optional-settings.md#git_ssl_no_verify) Environment Variable is set.
```
Comment (glennmatthews) [inline nautobot/docs/user-guides/git-data-source.md:82]:
```suggestion
    If you are using a self-signed Git repository, the Server Administrator will need to ensure the [`GIT_SSL_NO_VERIFY`](../configuration/optional-settings.md#git_ssl_no_verify) environment variable is set to permit this.
```

File: nautobot/docs/configuration/optional-settings.md (hunk 5 above)
Comment (smk4664) [reply nautobot/docs/configuration/optional-settings.md:374]: Correct.

**Review by glennmatthews — APPROVED** [review]
Thank you!

**Pull request merged**